py_ecc/bn128_curve.py

```diff
@@ -0,0 +1,180 @@
+from typing import (
+    cast,
+)
+
+from py_ecc.BaseCurve import (
+    BaseCurve,
+    BaseOptimizedCurve,
+)
+
+from py_ecc.curve_properties import (
+    curve_properties,
+    optimized_curve_properties,
+)
+
+from py_ecc.field_elements import (
+    FQP,
+    FQ12,
+)
+
+from py_ecc.field_properties import (
+    field_properties,
+)
+
+from py_ecc.optimized_field_elements import (
+    FQP as optimized_FQP,
+    FQ12 as optimized_FQ12,
+)
+
+from py_ecc.typing import (
+    Field,
+    Optimized_Point3D,
+    Point2D,
+)
+
+
+class BN128_Curve(BaseCurve):
+    curve_name = "bn128"
+    curve_order = curve_properties[curve_name]["curve_order"]
+    field_modulus = field_properties[curve_name]["field_modulus"]
+    b = curve_properties[curve_name]["b"]
+    b2 = curve_properties[curve_name]["b2"]
+    b12 = curve_properties[curve_name]["b12"]
+    G1 = curve_properties[curve_name]["G1"]
+    G2 = curve_properties[curve_name]["G2"]
+    Z1 = curve_properties[curve_name]["Z1"]
+    Z2 = curve_properties[curve_name]["Z2"]
+    ate_loop_count = curve_properties[curve_name]["ate_loop_count"]
+    log_ate_loop_count = curve_properties[curve_name]["log_ate_loop_count"]
+    pseudo_binary_encoding = curve_properties[curve_name]["pseudo_binary_encoding"]
+
+    def __init__(self) -> None:
+        super().__init__()
```

Comment:
This looks like it can be removed since it is a direct call to `super()` and thus doesn't change the base implementation.